state_machine/applications/system-check/tests/uart_test.py

```diff
@@ -0,0 +1,21 @@
+from lib.pycubed import cubesat
+import time
+
+async def run(result_dict):
+    cubesat.cam_pin.value = True
+    confirmed = False
+    st = time.monotonic()
+    buffer = bytearray(1)
+    while not confirmed:
+        if time.monotonic() - 10 > st:
+            result_dict["UART"] = ("timeout", False)
+            break
+
+        cubesat.uart.readinto(buffer)
+        if buffer[0] == 0xAA:
+            result_dict["UART"] = ("camera is on and running correct script", True)
+            confirmed = True
+        elif buffer[0] != 0xAA and buffer != 0x00:
+            result_dict["UART"] = ("camera is on but may be running the wrong script", False)
```

Comment:
We should probably output 3 things
if the uart driver loads without errors, like in the imu test:
```python3
if not cubesat.imu:
        result_dict["IMU"] = ("Not detected", False)
        return
    else:
        result_dict["IMU"] = ("Detected", True)
```
If we received a result from uart.
And finally if we received the correct result from uart, indicating the camera is running the correct script.

This help us figure out at which stage the uart test failed.